File: Data Structures & Algorithms/validate-parentheses/submission-1.cpp
class Solution {
public:
    bool isValid(string s) {
       stack<char> p;
       for(char c : s){
        if(c == '(' || c == '{' || c == '['){
            p.push(c);
        }else if ((c == '}' && p.size() > 0 && p.top() == '{') || 
                  ( c == ')' && p.size() > 0 && p.top() == '(') ||
                  (c == ']' && p.size() > 0 && p.top() == '[') ){
                  p.pop();
                  }
                  else {
                    p.push(c);
                  }
       } 

       return p.empty();
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {

    public record Temp(Integer temperature, Integer index) {}

    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] ans = new int[n];
        Stack<Temp> stk  = new Stack<>();
         for(int i = n-1 ; i >=0 ; i--){
            while( !stk.isEmpty() && temperatures[i] >= stk.peek().temperature()){
                stk.pop();
            }
            ans[i] = stk.empty() ? 0 : stk.peek().index() - i;
            stk.push(new Temp(temperatures[i],i));
         }
         return ans ;
    }
}
